lms/static/js/ccx/schedule.js

```diff
@@ -94,11 +111,19 @@ var edx = edx || {};
 	      if (vertical_location !== 'all') {
 		var chapter = self.chapter_select.val(),
 		    sequential = self.sequential_select.val();
-		var vertical = self.find_unit(
+		self.find_unit(
 		  self.hidden, chapter, sequential, vertical_location);
-		self.set_datetime('start', vertical.start);
-		self.set_datetime('due', vertical.due);
-	      }
+		// on selecting a unit: Disabled all date input fields.
+		// Because unit inherit dates from subsection.
+		self.disabled_fields($('.ccx_due_date_time_fields'), true);
+		self.disabled_fields($('.ccx_start_date_time_fields'), true);
+		    } else {
+		// On selecting "All units": Enabled all date input fields.
+		// Because unit inherit dates from subsection. And we are showing
+		// dates from selected subsection.
```

Comment:
```
// When "All units" is selected, all date fields are enabled,
// because units inherit dates from subsections and we
// are showing dates from the selected subsection. 
```